src/overlays/actors/ovl_En_Baba/z_en_baba.c

```diff
@@ -311,357 +336,364 @@ void func_80BA8DF4(EnBaba* this, PlayState* play) {
-                                &timePathTargetPos, this->timePathTimeSpeed);
-        SubS_TimePathing_ComputeInitialY(play, this->timePath, this->timePathWaypoint, &timePathTargetPos);
-        this->actor.world.pos.y = timePathTargetPos.y;
-        this->unk_438 = 1;
+                                this->timePathWaypointTime, this->timePathTotalTime, &this->timePathWaypoint,
+                                weightArray, &timePathPoint, this->timePathTimeSpeed);
+        SubS_TimePathing_ComputeInitialY(play, this->timePath, this->timePathWaypoint, &timePathPoint);
+        this->actor.world.pos.y = timePathPoint.y;
+        this->timePathIsSetup = true;
     } else {
-        timePathTargetPos = this->timePathTargetPos;
+        timePathPoint = this->timePathPoint;
     }
 
-    this->actor.world.pos.x = timePathTargetPos.x;
-    this->actor.world.pos.z = timePathTargetPos.z;
+    this->actor.world.pos.x = timePathPoint.x;
+    this->actor.world.pos.z = timePathPoint.z;
 
     if (SubS_InCsMode(play)) {
-        sp54 = this->timePathElapsedTime;
-        sp50 = this->timePathWaypoint;
-        timePathTargetPos = this->actor.world.pos;
+        prevTimePathElapsedTime = this->timePathElapsedTime;
+        prevTimePathWaypoint = this->timePathWaypoint;
+        timePathPoint = this->actor.world.pos;
     }
 
-    this->timePathTargetPos = gZeroVec3f;
+    this->timePathPoint = gZeroVec3f;
 
     if (SubS_TimePathing_Update(this->timePath, &this->timePathProgress, &this->timePathElapsedTime,
-                                this->timePathWaypointTime, this->timePathTotalTime, &this->timePathWaypoint, knots,
-                                &this->timePathTargetPos, this->timePathTimeSpeed)) {
-        this->unk_43C = 1;
+                                this->timePathWaypointTime, this->timePathTotalTime, &this->timePathWaypoint,
+                                weightArray, &this->timePathPoint, this->timePathTimeSpeed)) {
+        this->timePathHasReachedEnd = true;
     } else {
-        sp70 = this->actor.world.pos;
-        sp64 = this->timePathTargetPos;
-        this->actor.world.rot.y = Math_Vec3f_Yaw(&sp70, &sp64);
+        worldPos = this->actor.world.pos;
+        timePathPointNew = this->timePathPoint;
+        this->actor.world.rot.y = Math_Vec3f_Yaw(&worldPos, &timePathPointNew);
     }
 
     if (SubS_InCsMode(play)) {
-        this->timePathElapsedTime = sp54;
-        this->timePathWaypoint = sp50;
-        this->timePathTargetPos = timePathTargetPos;
+        this->timePathElapsedTime = prevTimePathElapsedTime;
+        this->timePathWaypoint = prevTimePathWaypoint;
+        this->timePathPoint = timePathPoint;
     }
 
     return false;
 }
 
-void func_80BA93AC(EnBaba* this, PlayState* play) {
-    if (this->unk_434 != 1) {
-        if (this->unk_434 == 2) {
+void EnBaba_HandleScheduleOutput(EnBaba* this, PlayState* play) {
```

Comment:
I wonder if `EnBaba_HandleScheduleOutput` and `EnBaba_ProcessScheduleOutput` should be `EnBaba_HandleScheduleResult` and `EnBaba_ProcessScheduleResult` respectively? Both are really only using the "result" part of the output. But then also if we want consistency between different actors that use the scheduling system, do they tend to use all of "scheduleOutput" in handling/processing?

Reply:
I think that makes sense for `EnBaba_HandleScheduleOutput`, and I would even shorten it more to `EnBaba_HandleSchedule` since it actually doesn't deal with the `ScheduleOutput` strcut. For `EnBaba_ProcessScheduleOutput`, on the other hand I think it makes more sense to keep `ScheduleOutput` mainly because the full `ScheduleOutput` struct is passed in, and while that function only deals with the `ScheduleResult`, it calls functions that makes use of the other fields.